explainer.md

```diff
@@ -245,16 +436,35 @@ and Ubuntu don't support them at all).
 Limiting support to integers makes behavior more predictable, though we are considering
 whether it might be worth adding support for other characters or symbols in future.
 
-### Why isn't this API available from a Service Worker?
+### Couldn’t this be a declarative API, so it would work without JavaScript?
+It could be, yes. However, as badges may be shared across multiple documents, this could be kind of confusing (e.g. there is a <link rel="shortcut icon badge" href="/favicon.ico" badge="99"> in the head of a document, but it is being badged with 7 because another page was loaded afterwards). There is some discussion of this [here](https://github.com/WICG/badging/issues/1#issuecomment-485635068).
+
+### Why can’t this be used in the background from the ServiceWorker? (see #28 and #5)
+
+Realistically, using the badge from the background is the most important use case, so ideally, badges could be updated in the background via the service worker.
+
+However there is a non-trivial problem: We need a way to inform the user that the app is running.
+
+With push notifications, browsers enforce that a notification be shown, or the browser will stop the app from processing additional notifications. Things are not as simple with badging, as a site could always set its badge to the same value, which would give the user would no cue that a site is running (or has run) in the background (and could mine all the cryptocurrencies!).
```

Comment:
Perhaps link to https://github.com/w3c/push-api/issues/313 since there's no spec text requiring this, that is the bug to add said text.